modules/component/src/push/push.pipe.ts

```diff
@@ -0,0 +1,61 @@
+import { ChangeDetectorRef, NgZone, Pipe, PipeTransform } from '@angular/core';
+import { NextObserver, Observable, PartialObserver, pipe, Subject } from 'rxjs';
+import { distinctUntilChanged, map, tap, withLatestFrom } from 'rxjs/operators';
+import {
+  CdAware,
+  CoalescingConfig as NgRxPushPipeConfig,
+  remainHigherOrder,
+  STATE_DEFAULT,
+} from '../core';
+
+@Pipe({ name: 'ngrxPush', pure: false })
+export class PushPipe extends CdAware implements PipeTransform {
+  private renderedValue: any = STATE_DEFAULT;
+
+  configSubject = new Subject<NgRxPushPipeConfig>();
+  config$ = this.configSubject.asObservable().pipe(distinctUntilChanged());
+
+  constructor(cdRef: ChangeDetectorRef, ngZone: NgZone) {
+    super(cdRef, ngZone);
+    this.subscription = this.observables$.subscribe();
+  }
+
+  transform<T>(obj: null | undefined, config?: NgRxPushPipeConfig): null;
+  transform<T>(obj: Observable<T>, config?: NgRxPushPipeConfig): T;
+  transform<T>(
+    obs: Observable<T> | null | undefined,
+    config: NgRxPushPipeConfig = { optimized: true }
+  ): T | null | undefined {
+    this.configSubject.next(config);
+    this.observablesSubject.next(obs);
+    return this.renderedValue;
+  }
+
+  getConfigurableBehaviour<T>(): remainHigherOrder<T> {
+    return pipe(
+      withLatestFrom(this.config$),
+      map(([value$, config]) => {
+        return !config.optimized
```

Comment:
Just leaving a note to say we discussed this, and are going to leave it in the codebase for now.

Reply:
Added Comments